internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/controller/VeniceParentHelixAdminTest.java

```diff
@@ -1250,6 +1260,47 @@ private void testSupersetSchemaGenerationWithUpdateDefaultValue(ControllerClient
 
   }
 
+  private void testEnumSchemaEvolution(
+      ControllerClient parentControllerClient,
+      ControllerClient childControllerClient) {
+    String storeName = Utils.getUniqueString("test_store");
+    String owner = "test_owner";
+    String keySchemaStr = "\"long\"";
+    String valueSchemaWithEnumDefaultDefined = "{\n" + "  \"name\": \"EnumTestRecord\",\n"
+        + "  \"namespace\": \"com.linkedin.avro.fastserde.generated.avro\",\n" + "  \"type\": \"record\",\n"
+        + "  \"fields\": [\n" + "    {\n" + "      \"name\": \"testEnum\",\n" + "      \"type\": {\n"
+        + "        \"type\": \"enum\",\n" + "        \"name\": \"TestEnum\",\n"
+        + "        \"symbols\": [\"A\", \"B\", \"C\"],\n" + "        \"default\": \"A\"\n" + "      }\n" + "    }\n"
+        + "  ]\n" + "}";
+    String valueSchemaWithEnumEvolved = "{\n" + "  \"name\": \"EnumTestRecord\",\n"
+        + "  \"namespace\": \"com.linkedin.avro.fastserde.generated.avro\",\n" + "  \"type\": \"record\",\n"
+        + "  \"fields\": [\n" + "    {\n" + "      \"name\": \"testEnum\",\n" + "      \"type\": {\n"
+        + "        \"type\": \"enum\",\n" + "        \"name\": \"TestEnum\",\n"
+        + "        \"symbols\": [\"A\", \"B\", \"C\", \"D\", \"E\"],\n" + "        \"default\": \"A\"\n" + "      }\n"
+        + "    }\n" + "  ]\n" + "}";
+    // Create a store with value schema with enum default defined
+    parentControllerClient.createNewStore(storeName, owner, keySchemaStr, valueSchemaWithEnumDefaultDefined);
+    // Try to evolve the enum field in the value schema
+    StoreInfo store = parentControllerClient.getStore(storeName).getStore();
+    assertFalse(store.isEnumSchemaEvolutionAllowed());
+    SchemaResponse schemaResponse = parentControllerClient.addValueSchema(storeName, valueSchemaWithEnumEvolved);
+    assertTrue(schemaResponse.isError(), "Enum schema evolution should not be allowed by default.");
+    MultiSchemaResponse allValueSchemaResponse = childControllerClient.getAllValueSchema(storeName);
+    assertEquals(allValueSchemaResponse.getSchemas().length, 1);
+    // Enable enum schema evolution
+    UpdateStoreQueryParams updateStoreQueryParams = new UpdateStoreQueryParams();
+    updateStoreQueryParams.setEnumSchemaEvolutionAllowed(true);
+    parentControllerClient.updateStore(storeName, updateStoreQueryParams);
+    // Try to evolve the enum field in the value schema again
+    store = parentControllerClient.getStore(storeName).getStore();
+    assertTrue(store.isEnumSchemaEvolutionAllowed());
```

Comment:
Can you also add a check that checks if enum schema evolution config got enabled on the child colo too?

Reply:
sure